src/useAnimatedValue.js

```diff
@@ -0,0 +1,45 @@
+import React from 'react'
+import { useSprings } from 'react-spring'
+
+const KEY = '__KEY'
+
+export function useAnimatedValue(initialValueOrFn) {
+  const [values, set] = useAnimatedValues(
+    1, 
+    () => (typeof initialValueOrFn === 'function') ? initialValueOrFn() : initialValueOrFn
+  )
+
+  const setValue = React.useCallback(
+    x => { set(() => x) }, 
+    [set]
+  )
+
+  return [values[0], setValue]
+}
+
+export function useAnimatedValues(count, initialValueFn) {
+  const [values, set] = useSprings(
+    count, 
+    i => ensureObject(initialValueFn(i), KEY)
+  )
+
+  const setValues = React.useCallback( 
+    fn => {
+      set(i => ({ 
+        ...ensureObject(fn(i), KEY), 
+        immediate: true 
```

Comment:
Ik twijfel of deze omgewisseld moeten worden. Wil je de controle hebben bij gebruik van `useAnimatedValues` of moet het altijd `immediate` zijn?

Reply:
Moet dit trouwens ook niet bij `initialValueFn` worden gedaan?